src/lTerm_sequence_lwt4.ml

```diff
@@ -0,0 +1,242 @@
+(* OCaml promise library
+ * http://www.ocsigen.org/lwt
+ * Copyright (C) 2009 Jérémie Dimino
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, with linking exceptions;
+ * either version 2.1 of the License, or (at your option) any later
+ * version. See COPYING file for details.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *)
+
+exception Empty
+
+type 'a t = {
```

Comment:
Not sure how this will work. Will this ```'a t``` be ```=``` to ```'a Lwt_sequence.t```? Pls seem my comment on lTerm.ml above. How is this type equality established?

Reply:
We don't need the type equality for Lwt4 since Lwt_sequence isn't going to be present, right?

Reply:
Hmm. I think we do since we use ```Lwt_main.enter_iter_hooks``` which returns ```Lwt_sequence.t```. My initial attempt got stuck here since I couldn't establish equality between the two types.